daemon/gamemode-context.c

```diff
@@ -801,6 +910,9 @@ static void *game_mode_context_reaper(void *userdata)
 			return NULL;
 		}
 
+		/* Check on the CPU/iGPU energy balance */
+		game_mode_check_igpu_energy(self);
```

Comment:
This one is subtle and probably deserves a comment somewhere...  The way this works in RAPL is that the values being queried aren't an instantaneous power in Watts, it's microjoules of energy used since boot (I think) or since the last time the counter rolled over.  The way you get average power in microwatts is by grabbing the energy before and after your window, subtracting, and dividing by the length of the window in seconds.  (You can then divide by 1000000 to get watts.)  Because we're only concerned with the ratio between the two values, we never bother dividing by the length of the window because that would just algebraically cancel out.  Instead, we just divide the GPU energy used in the window (difference of before and after) by the CPU energy used.  It all nicely averages and there are no instantaneous sampling problems.

Reply:
OK, A comment would be great just to keep this logged somewhere, but if we're already looking at an average then we're in the clear.

Reply:
I've added a comment at the place where we do the calculation.  That seemed like the most sensible place to add it.